src/features/profile/components/UserProfile/index.tsx

```diff
@@ -0,0 +1,54 @@
+'use client';
+import { IconPen } from '@/assets/images/icons';
+import { Button, Icon } from '@/components/ui';
+import { IUser } from '@/types/definition';
+
+import Image from 'next/image';
+import { useState } from 'react';
+
+import { RenameUserForm } from './RenameUserForm';
+
+interface IUserProfileProps {
+  /**
+   * The id of the user.
+   */
+  userId: IUser['id'];
+  /**
+   * The name of the user.
+   */
+  name: IUser['name'];
+}
+
+export const UserProfile = ({ userId, name }: IUserProfileProps) => {
+  const [isRenameUserFormOpen, setIsRenameUserFormOpen] = useState(false);
+
+  /**
+   * Open the rename user form when the pen icon is clicked.
+   */
+  const handlePenIconClick = () => {
+    setIsRenameUserFormOpen(true);
+  };
+
+  return (
+    <div className="flex flex-col items-center gap-4 pt-6">
+      <Image src="/images/avatar.png" width={128} height={128} alt="avatar" priority />
```

Comment:
`priority` property was necessary to avoid this caution ⬇️ . It seems like the size of the image is huge

<img width="710" alt="Screenshot 2024-03-07 at 0 03 18" src="https://github.com/nishiki-tech/nishiki-frontend/assets/99339182/2093185b-7dbf-44da-b0a9-fea656e65f6e">

Reference: https://nextjs.org/docs/pages/api-reference/components/image#priority

Reply:
My conclusion is either:

1. Using the SVG component as it is without `next/image` here.
2. Converting the SVG image into PNG and using it with `next/image` here.

But, in either case, place the image file under the `src/assets/images/ui/` directory, delete the old `icon_person-circle.svg` file, and replace all of them with the new file Also, using the `<Icon>` component should be avoided.

Usually, commonly we can import an SVG file as an "image" and specify it to `<Image src" ~ here ~ ">`. However, as you might have researched, we cannot take this approach because, in our project, all SVG images are automatically transformed into a React Component by `@svgr/webpack` being set up for the entire project.

### Approach 1 (SVG):
I prefer this because it's simpler.
```tsx
// SVG
<PersonCircle className="size-32" />
```

### Approach 2 (PNG):
This way is good when we care about the separation of concern. You can avoid defining image size using HTML attributes.
```tsx
// PNG
<figure className="size-32">
  <Image src={PersonCircle} fill />
</figure>
```

### Approach 3 (PNG):
Since we use Tailwind CSS, we are styling within HTML, which means the "separation of concern" is not completely achievable. So, you can simply define like so:
```tsx
// PNG
<Image src={PersonCircle} width="128" heigh="128" />
```

By the way, when you specify `<Image src" ~ this value ~ ">`, importing the image is better than manually specifying its path unless you need to dynamically define the `src` value. When you refactor or rename the image files, TypeScript can cause an error if you use imported images, while manual paths fail to display the images on the client side.